components/dashboard/src/components/access-control/access-control.tsx

```diff
@@ -447,4 +461,63 @@ export class AccessControl extends React.Component<AccessControlProps, AccessCon
             search: `returnTo=${returnTo}&host=${provider}&override=true&scopes=${scopes.join(',')}`
         }).toString();
     }
+
+    protected renderDisconnectDialog() {
+        const { disconnectDialog, user, authProviders } = this.state;
+        const authHost = disconnectDialog?.authHost;
+        const authProvider = authProviders.find(a => a.host === authHost);
+        if (!disconnectDialog || !user || !authProvider) {
+            return;
+        }
+
+        let message: JSX.Element;
+        const handleCancel = () => {
+            this.setState({
+                disconnectDialog: undefined
+            });
+        };
+        let buttonLabel: string;
+        let handleButton: () => void;
+
+        const thisUrl = new GitpodHostUrl(new URL(window.location.toString()));
+        const otherIdentitiesOfUser = user.identities.filter(i => i.authProviderId !== authProvider.authProviderId);
+        if (otherIdentitiesOfUser.length === 0) {
+            message = (<DialogContentText>
+                Disconnecting the single remaining provider would make your account unreachable. Please go the settings, if you want to delete the account.
```

Comment:
I'd rather not do this. Adding some extra cycles to avoid deleting by accident could be helpful in some cases. 

Also, this is something I'd like to push to a follow-up discussion and PR.

Reply:
DEAL 🤝

Reply:
Awesome! 🤝